server/config/service.go

```diff
@@ -133,6 +136,17 @@ func (s *service) newInitializedConfig(newStoredConfig StoredConfig) (*Config, u
 		localURL = "http://" + host + ":" + port + u.Path
 	}
 
+	// Generate an encryption key on the fly
+	// to encrypt/decrypt oauth user data
+	if conf.EncryptionKey == nil {
+		encKey, encErr := GenerateEncryptionKey()
+		if encErr != nil {
+			log.Errorf("Couldn't generate the encryption key for OAuth user data encryption")
+		}
+
+		conf.EncryptionKey = encKey
+	}
```

Comment:
I don't see that the `EncryptionKey` actually gets stored in the config file if no other event like `/apps settings` triggered the store process.

The steps to reproduce are:
1. Start server with empty `EncryptionKey`
2. Store stuff in OAuth store
3. Restart server
4. Confirm no `EncryptionKey` is stored (That is the issue!)